runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -78,6 +102,33 @@ public SparkPipelineResult run(Pipeline pipeline) {
     result.waitUntilFinish();
     assertThat(result, testPipelineOptions.getOnCreateMatcher());
     assertThat(result, testPipelineOptions.getOnSuccessMatcher());
+    // if the pipeline was executed in streaming mode, validate aggregators.
+    if (isStreaming) {
+      // validate assertion succeeded (at least once).
+      int success = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
```

Comment:
It would indeed be naive of me to expect any success to ingest data into the pipeline to also mean all data was successfully ingested, so I will add the same solution as `TestDataflowRunner`.
Opened [`BEAM-1173`](https://issues.apache.org/jira/browse/BEAM-1173) for a `Visitor`-based solution like you suggested.